charts/portworx/v2.8.1/templates/portworx-csi.yaml

```diff
@@ -65,32 +116,70 @@ spec:
       containers:
         - name: csi-external-provisioner
           imagePullPolicy: Always
-          image: "{{ template "px.getcsiProvisioner" . }}/csi-provisioner:v0.2.0"
+          image: {{ template "px.getCSIProvisionerImage" . }}
+          args:
+            - "--v=3"
+            {{- if (semverCompare "< 1.19.0" .Capabilities.KubeVersion.GitVersion) }}
+            - "--provisioner=pxd.portworx.com"
+            - "--enable-leader-election"
+            {{- end }}
+            - "--csi-address=$(ADDRESS)"
+            {{- if and (semverCompare ">=1.13.0" .Capabilities.KubeVersion.GitVersion) (semverCompare "<1.14.0" .Capabilities.KubeVersion.GitVersion) }}
+            - "--leader-election-type=leases"
+            {{- end }}
+          env:
+            - name: ADDRESS
+              value: /csi/csi.sock
+          securityContext:
+            privileged: true
+          volumeMounts:
+            - name: socket-dir
+              mountPath: /csi
+        - name: csi-snapshotter
+          imagePullPolicy: Always
+          image: {{ template "px.getCSISnapshotterImage" . }}
           args:
-            - "--v=5"
-            - "--provisioner=com.openstorage.pxd"
+            - "--v=3"
             - "--csi-address=$(ADDRESS)"
+            - "--leader-election=true"
+          env:
+            - name: ADDRESS
+              value: /csi/csi.sock
+          volumeMounts:
+            - name: socket-dir
+              mountPath: /csi
+        - name: csi-snapshot-controller
```

Comment:
We can only  include this sidecar for K8s 1.17 and later. 

equivalent px-installer code:
https://github.com/portworx/px-installer/blob/master/px-spec-websvc/csi.go#L172-L173

Reply:
updated